### PR TITLE
add short plugins docs, stolen from 15.04 appdev manual

### PR DESCRIPTION
**Pull request by dholbach**

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Extending Snapcraft: writing your own plugins
+
+Is Snapcraft missing support for your preferred build-system? Here's how you
+would add it!
+
+Snapcraft is extensible via plugins adding new part plugins. Let's add
```
Comment (elopio) [inline docs/plugins.md]: "via plugins adding new part plugins"
That sounds weird.

File: docs/plugins.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    crafty-target:
+        required: true
+```
+
+and `parts/plugins/x_crafty.py` with:
+
+import snapcraft
+
+class XCraftyPlugin(snapcraft.BasePlugin):
+    def build(self):
```
Comment (elopio) [inline docs/plugins.md]: one empty line after the class definition please.

**Comment by elopio** [comment]
lgtm. I've never tried this, but I've seen tests for this functionality. I will give it a try this week.

**Comment by kyrofa** [comment]
Thanks @dholbach, very useful addition! Other than @elopio's comments, looks good to me as well.

**Comment by kyrofa** [comment]
@dholbach sorry to be a bother, but can you please put the LP bug reference in the commit message on its own line, with no parentheses? Automated tools pull those out.

**Comment by dholbach** [comment]
Updated.

**Comment by kyrofa** [comment]
@elopio's comments still apply.

**Comment by elopio** [comment]
Too much talk, lets land :D
Thanks @dholbach.

**Pull request merged**